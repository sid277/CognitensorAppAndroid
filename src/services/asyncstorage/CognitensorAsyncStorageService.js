import AsyncStorage from '@react-native-community/async-storage';

class CognitensorAsyncStorageService {
  USER_TOKEN_KEY = 'USER_TOKEN';

  setUserToken = async (token) => {
    try {
      await AsyncStorage.setItem(this.USER_TOKEN_KEY, token);
    } catch (e) {
      console.warn('StorageClient: Failed to set user token');
    }
  };

  getUserToken = async () => {
    let userToken;
    try {
      userToken = await AsyncStorage.getItem(this.USER_TOKEN_KEY);
    } catch (e) {
      console.warn('StorageClient: Failed to get user token');
    }
    return userToken;
  };

  removeUserToken = async () => {
    try {
      await AsyncStorage.removeItem(this.USER_TOKEN_KEY);
    } catch (e) {
      console.warn('StorageClient: Failed to remove user token');
    }
  };
}

export default new CognitensorAsyncStorageService();
