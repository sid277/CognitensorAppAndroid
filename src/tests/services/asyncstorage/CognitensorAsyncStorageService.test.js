import React from 'react';
import { shallow } from 'enzyme';
import {CognitensorAsyncStorageService} from '../../../services/asyncstorage/CognitensorAsyncStorageService';
import renderer from 'react-test-renderer';


test('should test CognitensorAsyncStorageService component', () => {
const wrapper = CognitensorAsyncStorageService;
expect(wrapper).toMatchSnapshot();
});