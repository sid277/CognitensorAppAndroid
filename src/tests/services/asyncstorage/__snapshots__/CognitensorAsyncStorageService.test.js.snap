// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test CognitensorAsyncStorageService component 1`] = `undefined`;
